docs/buildrun.md

```diff
@@ -127,8 +146,9 @@ The following table illustrates the different states a BuildRun can have under i
 | Status | Reason | CompletionTime is set | Description |
 | --- | --- | --- | --- |
 | Unknown | Pending                       | No  | The BuildRun is waiting on a Pod in status Pending. |
+| Unknown | Running                       | No  | The BuildRun has been validate and started to perform its work. |l
 | Unknown | Running                       | No  | The BuildRun has been validate and started to perform its work. |
-| True    | Succeeded                     | Yes | The BuildRun Pod is done. |
+| Unknown | TaskRunCancelled              | No  | The user requested the BuildRun to be canceled.  This results in the BuildRun controller requesting the TaskRun be canceled.  Cancellation has not been done yet. |
```

Comment:
Our status reason should also be `"BuildRunCanceled"`. The code change might simply be extending https://github.com/shipwright-io/build/blob/ea7a88638532bd9601f4b6b5c0dbe93b09803199/pkg/reconciler/buildrun/resources/conditions.go#L43-L48

A nuance here would be to set `"BuildRunCanceling"` once we notice that the user sent us a request to cancel until the actual TaskRun got canceled and then set `"BuildRunCanceled"`. But that's an edge case differentiation that I do not insist on.

Reply:
OK so you concur with my speculation in this PR's description, and that the current form of the prototype is incomplete wrt what I showed there wrt `TaskRunCancelled` being shown.

Yes as I evolve this PR beyond WIP, assuming there are not dissenters to the API, I'll do this.